hopp/simulation/hopp.py

```diff
@@ -15,7 +15,7 @@
 class Hopp(BaseClass):
     name: str = field(converter=str)
     config: dict = field(converter=dict)
-    site: SiteInfo = field()
+    site: SiteInfo = field(converter=SiteInfo.from_dict)
```

Comment:
I like this! One similar issue I ran into for config data in `PVPlant` is that the type linter gives errors when you want to supply a `dict` to be converted because it doesn't match the type signature. A quick fix is to change the type signature to `Union[dict, SiteInfo]`